TYPO3.Flow/Classes/TYPO3/Flow/Mvc/Controller/Argument.php

```diff
@@ -148,6 +152,7 @@ public function getShortName()
      * Sets the data type of this argument that is also used for property mapping.
      * @param string $dataType
      * @return \TYPO3\Flow\Mvc\Controller\Argument $this
+     * @deprecated Will be removed for next major Flow version. Set the DataType via constructor.
```

Comment:
Well, in a current project the client uses exactly that, since the action has an abstract class as type hint. The concrete type to use is set in `initializeAction()` using this method.